msg/versioned/BatteryStatus.msg

```diff
@@ -1,74 +1,77 @@
-float32 current_a				# Battery current in amperes, -1 if unknown
-float32 current_average_a			# Battery current average in amperes (for FW average in level flight), -1 if unknown
-float32 discharged_mah				# Discharged amount in mAh, -1 if unknown
-float32 remaining				# From 1 to 0, -1 if unknown
-float32 scale					# Power scaling factor, >= 1, or -1 if unknown
-float32 time_remaining_s			# predicted time in seconds remaining until battery is empty under previous averaged load, NAN if unknown
-float32 temperature				# Temperature of the battery in degrees Celcius, NaN if unknown
-uint8 cell_count				# Number of cells, 0 if unknown
+uint64 timestamp # Time since system start [us]
+bool connected # Whether or not a battery is connected, based on a voltage threshold
+float32 voltage_v # Battery voltage [V] [@invalid 0]
+float32 current_a # Battery current [A] [@invalid -1]
+float32 current_average_a # Battery current average (for FW average in level flight) [A] [@invalid -1]
+float32 discharged_mah # Discharged amount [mAh] [@invalid -1]
+float32 remaining # Remaining capacity [@range 0,1] [@invalid -1]
+float32 scale # Power scaling factor [@range 1,] [@invalid -1]
+float32 time_remaining_s # Predicted time remaining until battery is empty under previous averaged load [s] [@invalid NaN]
+float32 temperature # Temperature of the battery [°C] [@invalid NaN]
+uint8 cell_count # Number of cells [@invalid 0]
+
 
-uint8 SOURCE_POWER_MODULE = 0
-uint8 SOURCE_EXTERNAL = 1
-uint8 SOURCE_ESCS = 2
-uint8 source					# Battery source
-uint8 priority					# Zero based priority is the connection on the Power Controller V1..Vn AKA BrickN-1
-uint16 capacity					# actual capacity of the battery
-uint16 cycle_count				# number of discharge cycles the battery has experienced
-uint16 average_time_to_empty			# predicted remaining battery capacity based on the average rate of discharge in min
-uint16 serial_number				# serial number of the battery pack
-uint16 manufacture_date				# manufacture date, part of serial number of the battery pack. Formatted as: Day + Month×32 + (Year–1980)×512
-uint16 state_of_health				# state of health. FullChargeCapacity/DesignCapacity, 0-100%.
-uint16 max_error				# max error, expected margin of error in % in the state-of-charge calculation with a range of 1 to 100%
-uint8 id					# ID number of a battery. Should be unique and consistent for the lifetime of a vehicle. 1-indexed.
-uint16 interface_error				# interface error counter
+uint8 source # Battery source [@enum SOURCE]
+uint8 SOURCE_POWER_MODULE = 0 # Power module
+uint8 SOURCE_EXTERNAL = 1 # External
+uint8 SOURCE_ESCS = 2 # ESCs
 
-float32[14] voltage_cell_v			# Battery individual cell voltages, 0 if unknown
-float32 max_cell_voltage_delta			# Max difference between individual cell voltages
+uint8 priority # Zero based priority is the connection on the Power Controller V1..Vn AKA BrickN-1
+uint16 capacity # Actual capacity of the battery [Ah]
```

Comment:
I copied units from MAVLink BUT it might actually be Wh, or mWh or something else. Do you know?

Reply:
It's mAh but wasn't documented before. I only found out by guessing Ah cannot be since it wouldn't be accurate enough as integer and then checking the code: https://github.com/PX4/PX4-Autopilot/commit/83d01a7c76eebbbec721a6e393a0b85ff1ef14f9#diff-53be6a45907694b5072a80ff35f2077c096abd82f452f6574dcac068cd6d448fR226
Note that this field is one of the we just dump some information somewhere which was added to the battery status. I plan to add a battery info message for such properties that don't change as quickly and don't need to be published at high rate and also double check what the fields are good for instead of just having one field per driver that provides some random additional information about the battery not overlapping with anything else.

Reply:
Ideally we'd allow for Wh calculations since they are generally more accurate for keeping track of energy.